dev/publicDev/scripts/resolver.js

```diff
@@ -3,23 +3,35 @@
 (function() {
   var app = angular.module('app');
 
-  app.service('resolver', customPageService);
+  app.service('resolver', service);
 
-  customPageService.$inject = ['$q', 'authService'];
+  service.$inject = ['$q', 'authService', 'UserCheckingService', '$location'];
 
-  function customPageService($q, authService) {
-    this.get = function () {
+  function service($q, authService, UserCheckingService, $location) {
+    this.get = function (action) {
+      return _.bind(function () {
+        return resolve(action);
+      }, this);
+    };
+    function resolve (action) {
       return  $q(function (resolve) {
-        authService.getCurrentUser().then(
-          function (res) {
-            authService.setUser(res.data);
-            resolve();
-          },
-          function (errRes) {
-            resolve();
-          }
+        Promise.all([authService.getCurrentUser(), authService.getRoles()]).then(
+            function (res) {
+              var user = res[0].data;
+              var roles = res[1].data;
+              authService.setUser(res[0].data);
+
+              if (action && !UserCheckingService.checkUser(user, roles, action)) {
+                $location.path('/');
+              }
+
+              resolve();
+            },
+            function (errRes) {
+              resolve();
```

Comment:
If user haven't loginned yet, we reseive the error as response on 'currentUser' reques.
This is resolver, it should be resolved even if error appears.